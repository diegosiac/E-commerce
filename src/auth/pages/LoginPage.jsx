import { useMemo } from 'react'
import { Link as RouterLink } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { useFormik } from 'formik'
import { object, string } from 'yup'
import { Alert, Grid, Link, TextField, Typography } from '@mui/material'
import { useAuthStore, useClearMessage } from '../../hooks'
import { startLogin } from '../../store/auth'
import { AuthLayout } from '../layout/AuthLayout'
import { InputLoding, InputPassword } from '../components'

const initialValues = {
  email: '',
  password: ''
}

const validationSchema = object({
  email: string()
    .required('El correo es requerido')
    .email('El correo no es válido'),
  password: string()
    .required('La contraseña es requerida')
    .min(6, 'Debe tener al menos 6 caracteres')
})

export const LoginPage = () => {
  const { status, errorMessage } = useAuthStore()
  const isCheckingAuthentication = useMemo(() => status === 'checking', [status])

  const dispatch = useDispatch()

  const createUser = (userData) => {
    dispatch(startLogin(userData))
  }

  useClearMessage()

  const { handleSubmit, handleChange, errors, values } = useFormik({
    initialValues,
    validationSchema,
    onSubmit: createUser
  })

  return (
    <AuthLayout title='Iniciar sesión'>
      <form aria-label='submit-form' onSubmit={handleSubmit}>
        <Grid container>
          <Grid item xs={12} className='mt-4'>
            <TextField
              name='email'
              type='email'
              label='Correo'
              placeholder='correo@google.com'
              value={values.email}
              onChange={handleChange}
              helperText={errors.email}
              color='success'
              autoComplete='email'
              aria-label='Ingresa tu correo'
              fullWidth
            />
          </Grid>

          <InputPassword onHandleChange={handleChange} value={values.password} error={errors.password} />

          <Grid container className='my-4 mt-1'>
            <Grid
              item
              xs={12}
              className={`${errorMessage ? '' : 'hidden'} mb-2`}
            >
              <Alert severity='error'>
                {errorMessage}
              </Alert>
            </Grid>

            <Grid item xs={12}>
              <InputLoding loading={isCheckingAuthentication} text='Iniciar Sesión' type='submit' />
            </Grid>
          </Grid>

          <Grid container direction='row' justifyContent='end'>
            <Typography className='mr-2'>¿Eres nuevo en GEEK MOBILE?</Typography>
            <Link
              component={RouterLink}
              color='inherit'
              to='/auth/register'
              title='Ir a Registrarse'
              aria-label='Ir a Registrarse'
            >
              Crear una cuenta
            </Link>
          </Grid>
        </Grid>
      </form>

    </AuthLayout>
  )
}
