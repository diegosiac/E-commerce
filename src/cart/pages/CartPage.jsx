import { Grid } from '@mui/material'
import { useCartStore } from '../../hooks'
import { Cart, PriceBox } from '../components'

export const CartPage = () => {
  const { basket } = useCartStore()

  const products = basket.filter(item => item.stock > 0)

  return (
    <Grid container component='div' className='justify-center my-10'>

      <Grid item className='flex gap-4 md:flex-nowrap flex-wrap w-11/12'>
        <Cart products={basket} />
        {
          products.length !== 0 &&
            <PriceBox products={products} />
        }
      </Grid>

    </Grid>
  )
}
