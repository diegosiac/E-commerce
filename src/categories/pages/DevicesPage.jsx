import { useProductStore } from '../../hooks'
import { CATEGORIES } from '../../helpers/categories'
import { CategoriesLayout } from '../layout/CategoriesLayout'
import { CardProduct, SkeletonProduct } from '../../product/components'
import devicesDigital from '../../assets/imgs/backgrounds/devicesDigital.webp'

export const DevicesPage = () => {
  const { products } = useProductStore()

  const componentsProducts = products?.filter(product => product.category === CATEGORIES.DEVICES)
  return (
    <CategoriesLayout
      title='Todos Nuestros Dispositivos'
      titlePrimary='Calidad de Nuestros Dispositivos '
      titleSecondary='Calidad superior'
      titleThird='Variedad de opciones'
      titleFourth='Soporte y servicio al cliente'
      descriptionPrimary='En nuestra empresa, nos enorgullece ofrecer dispositivos de la más alta calidad que se destacan por su excelencia y confiabilidad, tenemos una gran catidad de marcas de los mejores provedores'
      descriptionSecondary='Nos comprometemos a ofrecer dispositivos de la más alta calidad que cumplen y superan los estándares de la industria.'
      descriptionThird='Entendemos que cada cliente tiene necesidades y preferencias únicas. Por eso, ofrecemos una amplia gama de dispositivos'
      descriptionFourth='Nos comprometemos a brindar un excelente servicio al cliente en cada etapa del proceso.'
      urlSecondary={devicesDigital}
    >
      {
        products
          ? componentsProducts.map(({ id, ...props }) => (
            <CardProduct
              key={id}
              id={id}
              {...props}
            />
          ))
          : (
            <>
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
            </>
            )
      }
    </CategoriesLayout>
  )
}
