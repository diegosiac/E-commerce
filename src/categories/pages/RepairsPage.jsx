import { useProductStore } from '../../hooks'
import { CATEGORIES } from '../../helpers/categories'
import { CardProduct, SkeletonProduct } from '../../product/components'
import { CategoriesLayout } from '../layout/CategoriesLayout'
import smartphoneRepair from '../../assets/imgs/backgrounds/smartphoneRepair.webp'

export const RepairsPage = () => {
  const { products } = useProductStore()

  const componentsProducts = products?.filter(product => product.category === CATEGORIES.REPAIRS)
  return (
    <CategoriesLayout
      title='Todas Nuestras Reparaciónes'
      titlePrimary='Calidad de Nuestras Reparaciónes'
      titleSecondary='Experticia y conocimientos técnicos'
      titleThird='Uso de piezas de calidad'
      titleFourth='Garantía sólida'
      descriptionPrimary='Nuestro enfoque en la calidad comienza desde el primer contacto con nuestros clientes. Nuestro equipo de servicio al cliente está altamente capacitado para entender y evaluar los problemas que enfrentan tus dispositivos o equipos.'
      descriptionSecondary='Nuestro equipo de técnicos cuenta con una amplia experiencia y conocimientos en una variedad de dispositivos y equipos.'
      descriptionThird='Entendemos la importancia de utilizar piezas de repuesto de alta calidad en nuestras reparaciones.'
      descriptionFourth='Respaldamos nuestro trabajo con garantías sólidas. Queremos que te sientas seguro y satisfecho con nuestros servicios.'
      urlSecondary={smartphoneRepair}
    >
      {
        products
          ? componentsProducts.map(({ id, ...props }) => (
            <CardProduct
              key={id}
              id={id}
              {...props}
            />
          ))
          : (
            <>
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
              <SkeletonProduct />
            </>
            )
      }
    </CategoriesLayout>
  )
}
