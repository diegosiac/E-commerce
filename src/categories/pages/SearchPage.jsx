import { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { Navigate } from 'react-router-dom'
import { Grid, Typography } from '@mui/material'
import { useQuery } from '../../hooks'
import { getProductSearch } from '../../helpers'
import { setQuery } from '../../store/products/thunks'
import { CardProduct } from '../../product/components'
import { CategoriesLayout } from '../layout/CategoriesLayout'

export const SearchPage = () => {
  const dispatch = useDispatch()
  const queryRouter = useQuery()
  const [products, setProducts] = useState([])
  const queryRef = queryRouter.get('query')

  useEffect(() => {
    if (queryRef) {
      dispatch(setQuery(queryRef))
      setResultProducts(queryRef)
    }
  }, [queryRef])

  if (!queryRef) return <Navigate to='/' replace />

  const setResultProducts = async (query) => {
    const results = await getProductSearch(query)

    setProducts(results)
  }

  return (
    <CategoriesLayout
      title={`Resultados para "${queryRef}"`}
      disableGrid
    >
      {
        products.length !== 0
          ? products.map(({ id, ...props }) => (
            <CardProduct
              key={id}
              id={id}
              {...props}
            />
          ))
          : <Grid container className='bg-white justify-center items-center min-h-[400px] rounded-xl'>
            <Typography variant='h4' component='h2'>No se encontraron resultados</Typography>
            </Grid>
      }
    </CategoriesLayout>
  )
}
