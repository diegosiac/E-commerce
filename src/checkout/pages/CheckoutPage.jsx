import { useState } from 'react'
import { Grid, Paper, Step, StepLabel, Stepper, Typography } from '@mui/material'
import { AddressForm, Review } from '../components'
import { useCartStore } from '../../hooks'

const steps = ['Dirección de Envío', 'Resumen de compra']

export const CheckoutPage = () => {
  const { basket } = useCartStore()
  const [activeStep, setActiveStep] = useState(0)
  const [addressData, setAddressData] = useState({
    firstName: '',
    lastName: '',
    address1: '',
    phoneNumber: '',
    countryRegion: '',
    zip: '',
    state: '',
    locality: '',
    sublocality: ''
  })

  const handleNext = () => setActiveStep(activeStep + 1)

  const handleBack = () => setActiveStep(activeStep - 1)

  const products = basket.filter(item => item.stock > 0)

  const getStepContent = (step) => {
    switch (step) {
      case 0:
        return <AddressForm handleNext={handleNext} setData={setAddressData} addressData={addressData} />
      case 1:
        return <Review handleBack={handleBack} addressData={addressData} basket={products} />
      default:
        throw new Error('Unknown step')
    }
  }

  return (
    <Grid
      container
      className='justify-center'
    >
      <Paper variant='outlined' className='my-10 p-4 min-w-[370px] w-[75%]'>
        <Typography component='h1' variant='h2' className='text-center'>
          Proceso de pago
        </Typography>

        <Stepper activeStep={activeStep} className='py-6'>
          {steps.map(label => (
            <Step key={label}>
              <StepLabel>{label}</StepLabel>
            </Step>
          ))}
        </Stepper>

        {getStepContent(activeStep)}
      </Paper>
    </Grid>
  )
}
