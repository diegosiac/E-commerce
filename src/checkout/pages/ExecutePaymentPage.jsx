import { useEffect, useState } from 'react'
import { Navigate } from 'react-router-dom'
import { Grid, Typography } from '@mui/material'
import { useQuery } from '../../hooks'
import { getCheckoutOrder } from '../../helpers'
import { CheckoutStatusLayout } from '../layout/CheckoutStatusLayout'
import { DetailsShop, ItemProduct } from '../../purchases/components'
import { SkeletonCheckout } from '../components'

export const ExecutePaymentPage = () => {
  const orderId = useQuery().get('id')
  if (!orderId) return <Navigate to='/' replace />

  const [loading, setLoading] = useState(true)
  const [order, setOrder] = useState(null)

  useEffect(() => {
    getOrder()
  }, [])

  const getOrder = async () => {
    const order = await getCheckoutOrder({ orderId })

    setOrder(order)
    setLoading(false)
  }

  if (loading) return <SkeletonCheckout />

  if (order === undefined) return <Navigate to='/' replace />

  const { address, amount, id, dateShop, products, delivery } = order
  return (
    <CheckoutStatusLayout title='Gracias por tu compra' boxColor='bg-green-600'>

      <Grid container className='mb-8'>
        <Typography variant='h4' component='h3' className='mb-3'>Detalles de la Compra</Typography>
        <DetailsShop
          shippingAddress={address}
          totalValue={amount}
          value={amount}
          id={id}
          dateShop={dateShop}
          deliveryDay={delivery}
        />
      </Grid>

      <Grid container>
        <Typography variant='h4' component='h3' className='mb-3'>Art√≠culos Comprados</Typography>
        <Grid
          container
          className='flex-col gap-5 p-4 bg-white'
        >
          {
            products.map(({ _id, thumbnail, name, value, quantity }) => {
              return (
                <ItemProduct
                  key={_id}
                  name={name}
                  thumbnail={thumbnail}
                  value={value}
                  amount={quantity}
                />
              )
            })
          }
        </Grid>
      </Grid>

    </CheckoutStatusLayout>
  )
}
