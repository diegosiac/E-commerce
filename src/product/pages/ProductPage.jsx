import { Navigate } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { useProductStore, useQuery, useCartStore, useAuthStore } from '../../hooks'
import { setBasketProduct } from '../../store/cart/thunks'
import { ContendProductPage } from '../components'

export const ProductPage = () => {
  const { status: isAuth } = useAuthStore()
  const { products } = useProductStore()
  const { status, basket } = useCartStore()
  const dispatch = useDispatch()
  const querys = useQuery()

  const queryId = querys.get('id')

  if (!products) {
    return (
      <>
        <ContendProductPage isLoading />
      </>
    )
  }

  const isValidProduct = products.find(product => product.id === queryId)

  if (!isValidProduct) return <Navigate to='not-found' replace />

  const isAuthenticated = isAuth === 'authenticated'

  const { name, value, stock, thumbnail, description, id } = isValidProduct

  const isExistProductCart = basket.some(product => product.id_product === id)

  const textLink = name.trim().replace(/ /g, '-').replace(/\//g, '')

  window.history.pushState(null, '', `/product/${textLink}?id=${id}`)

  const handleAddToBasket = () => {
    if (isExistProductCart) return
    dispatch(setBasketProduct({ id }))
  }

  return (
    <>
      <ContendProductPage
        name={name}
        value={value}
        stock={stock}
        thumbnail={thumbnail}
        description={description}
        id={id}
        handleAddToBasket={handleAddToBasket}
        status={status}
        statusProduct={isExistProductCart}
        isAuthenticated={isAuthenticated}
      />
    </>
  )
}
