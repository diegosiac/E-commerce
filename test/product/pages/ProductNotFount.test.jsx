import { render } from '@testing-library/react'
import { ProductNotFount } from '../../../src/product/pages'

describe('Testing on <ProductNotFount/>', () => {
  test('must match the snapshot ', () => {
    const tree = render(
      <ProductNotFount />
    )

    expect(tree).toMatchSnapshot()
  })
})
