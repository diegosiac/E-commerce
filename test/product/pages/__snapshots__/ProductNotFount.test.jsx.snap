// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <ProductNotFount/> must match the snapshot  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container justify-center items-center bg-white w-10/12 rounded-md h-96 my-10 flex-col text-center text-gray-400 css-11lq3yg-MuiGrid-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium w-28 h-28 css-i4bv87-MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 470.333 470.333"
        >
          <g>
            <g>
              <path
                d="M235.167,470.333c129.767,0,235.167-105.4,235.167-235.167S364.933,0,235.167,0S0,105.4,0,235.167    S105.4,470.333,235.167,470.333z M235.167,34c110.783,0,201.167,90.383,201.167,201.167S345.95,436.333,235.167,436.333    S34,345.95,34,235.167S124.383,34,235.167,34z"
              />
              <path
                d="M164.333,196.633c7.083,0,13.883-2.833,18.983-7.933c10.483-10.483,10.483-27.483,0-37.967    c-5.1-5.1-11.9-7.933-18.983-7.933s-13.883,2.833-18.983,7.933c-10.483,10.483-10.483,27.483,0,37.967    C150.45,193.8,157.25,196.633,164.333,196.633z"
              />
              <path
                d="M299.2,196.633c7.083,0,13.883-2.833,18.983-7.933c10.483-10.483,10.483-27.483,0-37.967    c-5.1-5.1-11.9-7.933-18.983-7.933s-13.883,2.833-18.983,7.933c-10.483,10.483-10.483,27.483,0,37.967    C285.317,193.8,291.833,196.633,299.2,196.633z"
              />
              <path
                d="M157.25,328.667c0.567-1.7,11.617-38.25,73.667-38.25c64.033,0,82.167,38.817,82.733,40.233    c2.833,6.517,9.067,10.483,15.583,10.483c2.267,0,4.25-0.283,6.517-1.417c8.783-3.683,12.75-13.6,9.067-22.1    c-1.133-2.55-26.35-61.2-114.183-61.2c-89.533,0-105.967,61.2-106.533,63.75l16.433,4.25L157.25,328.667z"
              />
            </g>
          </g>
        </svg>
        <h2
          class="MuiTypography-root MuiTypography-h1 text-inherit mt-5 css-o2w69a-MuiTypography-root"
        >
          404
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h3 text-inherit my-2 css-gepadz-MuiTypography-root"
        >
          Producto no encontrado
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h6 text-inherit css-2ulfj5-MuiTypography-root"
        >
          El producto no existe o ya no hay disponibilidad
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container justify-center items-center bg-white w-10/12 rounded-md h-96 my-10 flex-col text-center text-gray-400 css-11lq3yg-MuiGrid-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium w-28 h-28 css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 470.333 470.333"
      >
        <g>
          <g>
            <path
              d="M235.167,470.333c129.767,0,235.167-105.4,235.167-235.167S364.933,0,235.167,0S0,105.4,0,235.167    S105.4,470.333,235.167,470.333z M235.167,34c110.783,0,201.167,90.383,201.167,201.167S345.95,436.333,235.167,436.333    S34,345.95,34,235.167S124.383,34,235.167,34z"
            />
            <path
              d="M164.333,196.633c7.083,0,13.883-2.833,18.983-7.933c10.483-10.483,10.483-27.483,0-37.967    c-5.1-5.1-11.9-7.933-18.983-7.933s-13.883,2.833-18.983,7.933c-10.483,10.483-10.483,27.483,0,37.967    C150.45,193.8,157.25,196.633,164.333,196.633z"
            />
            <path
              d="M299.2,196.633c7.083,0,13.883-2.833,18.983-7.933c10.483-10.483,10.483-27.483,0-37.967    c-5.1-5.1-11.9-7.933-18.983-7.933s-13.883,2.833-18.983,7.933c-10.483,10.483-10.483,27.483,0,37.967    C285.317,193.8,291.833,196.633,299.2,196.633z"
            />
            <path
              d="M157.25,328.667c0.567-1.7,11.617-38.25,73.667-38.25c64.033,0,82.167,38.817,82.733,40.233    c2.833,6.517,9.067,10.483,15.583,10.483c2.267,0,4.25-0.283,6.517-1.417c8.783-3.683,12.75-13.6,9.067-22.1    c-1.133-2.55-26.35-61.2-114.183-61.2c-89.533,0-105.967,61.2-106.533,63.75l16.433,4.25L157.25,328.667z"
            />
          </g>
        </g>
      </svg>
      <h2
        class="MuiTypography-root MuiTypography-h1 text-inherit mt-5 css-o2w69a-MuiTypography-root"
      >
        404
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h3 text-inherit my-2 css-gepadz-MuiTypography-root"
      >
        Producto no encontrado
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h6 text-inherit css-2ulfj5-MuiTypography-root"
      >
        El producto no existe o ya no hay disponibilidad
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
